app/src/main/kotlin/compiler/intermediate_form/ControlFlowGraphBuilder.kt

```diff
@@ -0,0 +1,42 @@
+package compiler.intermediate_form
+
+import compiler.common.reference_collections.ReferenceHashMap
+
+class ControlFlowGraphBuilder {
```

Comment:
Would you mind adding some tests for this class? It would be a good way of documenting the properties of the `ControlFlowGraph`s and merging operations performed on them.